syntax = "proto3";

package pyc.chat;

message AddFriendReq {
    int32 apply_uid = 1;
    string name = 2;
    string desc = 3;
    int32 to_uid = 4;
}

message AddFriendRsp {
    int32 error = 1;
    int32 apply_uid = 2;
    int32 to_uid = 3;
}

message ReplyFriendReq {
    int32 reply_uid = 1;
    bool agree = 2;
    int32 to_uid = 3;
}

message ReplyFriendRsp {
    int32 error = 1;
    int32 reply_uid = 2;
    int32 to_uid = 3;
}

message SendChatMsgReq {
    int32 from_uid = 1;
    int32 to_uid = 2;
    string msg = 3;
}

message SendChatMsgRsp {
    int32 error = 1;
    int32 from_uid = 2;
    int32 to_uid = 3;
}

message AuthFriendReq {
    int32 from_uid = 1;
    int32 to_uid = 2;
}

message AuthFriendRsp {
    int32 error = 1;
    int32 from_uid = 2;
    int32 to_uid = 3;
}

message TextChatData {
    string msg_id = 1;
    string msg_content = 2;
}

message TextChatMsgReq {
    int32 from_uid = 1;
    int32 to_uid = 2;
    repeated TextChatData text_msgs = 3;
}

message TextChatMsgRsp {
    int32 error = 1;
    int32 from_uid = 2;
    int32 to_uid = 3;
    repeated TextChatData text_msgs = 4;
}

service ChatService {
    rpc NotifyAddFriend (AddFriendReq) returns (AddFriendRsp);
    rpc ReplyAddFriend (ReplyFriendReq) returns (ReplyFriendRsp);
    rpc SendChatMsg (SendChatMsgReq) returns (SendChatMsgRsp);
    rpc NotifyAuthFriend (AuthFriendReq) returns (AuthFriendRsp);
    rpc NotifyTextChatMsg (TextChatMsgReq) returns (TextChatMsgRsp);
}